tests/fullstack-test/mpp/remote_read_virtual_column.test

```diff
@@ -0,0 +1,41 @@
+# Copyright 2025 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Preparation.
+=> DBGInvoke __init_fail_point()
+
+mysql> drop table if exists test.test_vir;
+mysql> create table test.test_vir (id int not null, str varchar(100), str_vir varchar(100) as (lower(str)) VIRTUAL);
+mysql> insert into test.test_vir (id, str) values(1, 'ABC'), (2, 'Def'), (3, 'xXXX');
+mysql> alter table test.test_vir set tiflash replica 1;
+mysql> set @@cte_max_recursion_depth = 10000000; insert into test.test_vir (id, str) with recursive cte1 as (select 1 c1, 'a' c2 union select c1+1 c1, substr(concat(c2, 'B'), 10) c2 from cte1 limit 10000) select * from cte1;
+
+func> wait_table test test_vir
+mysql> split table test.test_vir by (100), (500), (1000);
+TOTAL_SPLIT_REGION	SCATTER_FINISH_RATIO
+3	1
+
+=> DBGInvoke __enable_fail_point(force_random_remote_read_for_partition_table)
+mysql> set @@tidb_isolation_read_engines='tiflash'; set @@tidb_enforce_mpp=1; select * from test.test_vir order by 1, 2, 3 limit 5;
```

Comment:
It seems that the table must have at least 2 regions, and part of them trigger remote read, while others do not.